policy/scan/local_scanner.go

```diff
@@ -545,3 +555,32 @@ func (s *localAssetScanner) UpdateFilters(filters *policy.Mqueries, timeout time
 
 	return queries, err
 }
+
+func setPlatforms(ctx context.Context, assetList []*asset.Asset, im inventory.InventoryManager, record bool) error {
+	for i := range assetList {
+		a := assetList[i]
+		if a.Platform.Kind == providers.Kind_KIND_CONTAINER_IMAGE {
```

Comment:
Nit: can we put the kind in a list, so building upon in is easier?

Reply:
This makes the scan much slower because we pull that image twice right?

Reply:
Yes, we pull the image twice now. One option to not pull twice is to re-use the connection from this function for the actual scan. However, that would cause memory issues when running this code in a container. We had that before and the container ended up using gigabytes of memory because it was stacking images on the disk.

The other option is to call `SynchronizeAssets` again after the scan is run for every asset that is affected. I think there was the intention to do this before since there is a commented out line in the code right now that is tracking assets that need to be re-synced